components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/ApisApiServiceImpl.java

```diff
@@ -908,83 +929,99 @@ public Response apisApiIdUserRatingDelete(String apiId, String xWSO2Tenant, Stri
-            for (Environment environment: environments) {
-               if (environment.getName().equals(environmentName)) {
-                   selectedEnvironment = environment;
-                   break;
-               }
-            }
+        try {
+            APIConsumer apiConsumer = RestApiCommonUtil.getLoggedInUserConsumer();
+            String orgId = RestApiCommonUtil.getOrgIdMatchesTenantDomain(organizationId, xWSO2Tenant);
+            OrganizationDTO organizationDTO = APIUtil.getOrganizationDTOFromOrgID(orgId, xWSO2Tenant);
+            API api = apiConsumer.getLightweightAPIByUUID(apiId, organizationDTO);
+            APIIdentifier apiIdentifier = api.getId();
+
+            List<Environment> environments = APIUtil.getEnvironmentsOfAPI(api);
+            if (environments != null && environments.size() > 0) {
+                if (StringUtils.isEmpty(labelName) && StringUtils.isEmpty(environmentName)) {
+                    environmentName = api.getEnvironments().iterator().next();
+                }
 
-            if (selectedEnvironment == null) {
-                throw new APIManagementException(ExceptionCodes.from(ExceptionCodes.GATEWAY_ENVIRONMENT_NOT_FOUND,
-                        environmentName));
-            }
-            ResourceFile wsdl = apiConsumer.getWSDL(api, selectedEnvironment.getName(), selectedEnvironment.getType(),
-                    requestedTenantDomain);
+                Environment selectedEnvironment = null;
+                for (Environment environment: environments) {
+                    if (environment.getName().equals(environmentName)) {
+                        selectedEnvironment = environment;
+                        break;
+                    }
+                }
 
-            return RestApiUtil.getResponseFromResourceFile(apiIdentifier.toString(), wsdl);
-        } else {
-            throw new APIManagementException(ExceptionCodes.from(ExceptionCodes.NO_GATEWAY_ENVIRONMENTS_ADDED,
-                    apiIdentifier.toString()));
+                if (selectedEnvironment == null) {
+                    throw new APIManagementException(ExceptionCodes.from(ExceptionCodes.GATEWAY_ENVIRONMENT_NOT_FOUND,
+                            environmentName));
+                }
+                ResourceFile wsdl = apiConsumer.getWSDL(api, selectedEnvironment.getName(), selectedEnvironment.getType(),
+                        orgId);
+
+                return RestApiUtil.getResponseFromResourceFile(apiIdentifier.toString(), wsdl);
+            } else {
+                throw new APIManagementException(ExceptionCodes.from(ExceptionCodes.NO_GATEWAY_ENVIRONMENTS_ADDED,
+                        apiIdentifier.toString()));
+            }
+        } catch (UserStoreException e) {
+            String errorMessage = "Error while checking availability of tenant " +xWSO2Tenant;
+            RestApiUtil.handleInternalServerError(errorMessage, e, log);
         }
```

Comment:
I think its better not to throw a UserStoreException from RestApiCommonUtil.getOrgIdMatchesTenantDomain. You can catch UserStoreException inside that method and throw an APIManagementException with a proper [ExceptionCode](https://github.com/wso2/carbon-apimgt/wiki/Exception-handling-in-REST-APIs). Then just throw the APIManagementException from this method too. REST API can automatically handle it.

Reply:
Fixed https://github.com/wso2/carbon-apimgt/pull/9609/commits/629fd2c956b66218c6f5b9d125e4b0f25fe735e4